### PR TITLE
Wip/walk prop - dev experiment

### PR DESCRIPTION
**Pull request by bewest**
Experiment to handle type of incoming parameters.

### CODE REVIEW

File: lib/entries.js [hunk 1]
```diff
@@ -10,17 +11,32 @@ var ObjectID = require('mongodb').ObjectID;
...
-  if (opts && opts.find && opts.find.sgv) {
-    Object.keys(opts.find.sgv).forEach(function (key) {
-      var is_keyword = /^\$/g;
-      if (is_keyword.test(key)) {
-        opts.find.sgv[key] = parseInt(opts.find.sgv[key]);
-      }
-    });
-  }
+
+  opts = walk_prop('date', parseInt)(opts);
```
Comment (jasoncalabrese) [inline lib/entries.js]: an array to list these int fields fields would be nice, maybe a map?

```
var typedProps = [
  date: parseInt
  , sgv: parseInt
];

**Comment by bewest** [comment]
All done, this should probably move to it's own module, any opinion there?

**Comment by jasoncalabrese** [comment]
yeah, would be nice to be able to reuse for the treatments api, easier to test then too

**Comment by bewest** [comment]
Where to put it? In `storage.js`?  In `lib/foobar`?  In `lib/utils.js`?  `lib/` feels like big directory.

**Comment by jasoncalabrese** [comment]
maybe in something like `/lib/api/walker.js`

**Comment by jasoncalabrese** [comment]
guess it's more generic than that, maybe just adding it to utils?

**Pull request merged**